backend/canisters/notifications_index/impl/src/model/fcm_token_store.rs

```diff
@@ -0,0 +1,116 @@
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeSet;
+use types::{FcmToken, UserId};
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct FcmTokenStore {
+    // User tokens are stored in a BTreeSet to ensure uniqueness!
+    fcm_user_tokens: BTreeSet<(UserId, FcmToken)>,
+}
+
+impl FcmTokenStore {
+    /// Checks if a given FCM token exists!
+    pub fn check_token_exists(&self, token: &FcmToken) -> bool {
+        self.fcm_user_tokens.iter().any(|(_, t)| t == token)
+    }
+
+    /// Add a new FCM token, and assign it to a user.
+    ///
+    /// If the token already exists, it will not be added again - this also
+    /// ensures that the user does not have duplicate tokens, or that the same
+    /// token is not assigned to multiple users.
+    pub fn add_token(&mut self, user_id: UserId, token: FcmToken) -> Result<(), String> {
```

Comment:
The method names should probably be `exists` (or maybe `contains`?), `add`, `remove`, and maybe `get_all`.
The type name is `FcmTokenStore` so having `token` in each method name seems redundant.